src/package-manager.ts

```diff
@@ -70,6 +78,19 @@ export const packageManagers: PackageManager[] = [
   },
 ] as const;
 
+function parsePackageManagerName(name: string, disableWarn: boolean): any {
+  const sanitized = name.replace(/^\W+/, "");
+  if (name !== sanitized) {
+    if (!disableWarn) {
+      consola.warn(
```

Comment:
API utils don't show warning (or depend on consola) we might return an object and pass it to final resolve object (something like `abnormalChars: true` or `warnings: []` but no console warn here)